app/models/taxonomy.rb

```diff
@@ -28,6 +28,10 @@ class Taxonomy < ApplicationRecord
   has_many :subnets, :through => :taxable_taxonomies, :source => :taxable, :source_type => 'Subnet'
   has_many :auth_sources, :through => :taxable_taxonomies, :source => :taxable, :source_type => 'AuthSource'
 
+  has_many :puppetclasses, :through => :environments
```

Comment:
Should this also have the other side on the puppetclass side?

Reply:
The associations were not necessary in the case of puppetclasses to fix the API, but they are added now.